File: LuckyBol/app/src/main/java/com/pplus/luckybol/core/network/service/MapService.java
package com.pplus.luckybol.core.network.service;


import com.pplus.luckybol.LuckyBolApplication;
import com.pplus.luckybol.R;
import com.pplus.luckybol.core.network.apis.IMapApi;

import com.pplus.networks.core.NetworkHeader;
import com.pplus.networks.service.AbstractService;
import okhttp3.OkHttpClient;
import retrofit2.Converter;

/**
 * Created by J2N on 16. 6. 20..
 */
public class MapService extends AbstractService<IMapApi> {

    @Override
    public String getEndPoint(){

        //map base url
        return "https://dapi.kakao.com/";
    }

    @Override
    public OkHttpClient getCustomHttpClient(){

        return null;
    }

    @Override
    public Converter.Factory getCustomConverterFactory(){

        return null;
    }

    @Override
    public void initializeHeader(NetworkHeader header){

                header.addHeader("Authorization", "KakaoAK "+ LuckyBolApplication.getContext().getString(R.string.kakao_rest_key));
        //        header.addHeader("HTTP_CLIENT_ACCT", NetworkUtil.getLuhnNuber("user_key"));
    }
}

File: PRNumber/app/src/main/java/com/pplus/prnumberuser/core/network/service/MapService.kt
package com.pplus.prnumberuser.core.network.service

import com.pplus.networks.core.NetworkHeader
import com.pplus.networks.service.AbstractService
import com.pplus.prnumberuser.PRNumberApplication
import com.pplus.prnumberuser.R
import com.pplus.prnumberuser.core.network.apis.IMapApi
import okhttp3.OkHttpClient
import retrofit2.Converter

/**
 * Created by J2N on 16. 6. 20..
 */
class MapService : AbstractService<IMapApi?>() {
    override fun getEndPoint(): String {

        //map base url
        return "https://dapi.kakao.com/"
    }

    override fun getCustomHttpClient(): OkHttpClient? {
        return null
    }

    override fun getCustomConverterFactory(): Converter.Factory? {
        return null
    }

    override fun initializeHeader(header: NetworkHeader) {
        header.addHeader("Authorization", "KakaoAK " + PRNumberApplication.getContext().getString(R.string.kakao_rest_key)) //        header.addHeader("HTTP_CLIENT_ACCT", NetworkUtil.getLuhnNuber("user_key"));
    }
}